Add ToggleLockBranch (#14)
SonarSource/gh-action-lt-backlog: ToggleLockBranch/ReadMe.md

# ToggleLockBranch

Modify branch protection rules to Lock or Unlock configured branch.

Locked branches cannot be modified, preventing pull request from being merged.

> Intended usage is to prevent any changes on a repository that is about to be released. And development team needs to lock it temporarily.

Reviewer: ```suggestion
The intended usage is to prevent any changes on a repository that is about to be released when the development team needs to lock it temporarily.
```

## Inputs

### `github-token`

> Token to access GitHub API.

Reviewer: ```suggestion
Token to access the GitHub API.
```

### `branch-pattern`

> Pattern of the branch protection rule that should be locked or unlocked.

Reviewer: ```suggestion
The pattern of the branch protection rule that should be locked or unlocked.
```

This parameter is optional. 

Default value: `master`

## Outputs

None

## Prerequisites

Send email to REs - ask to add a `github_lock` token to the Vault configuration of your repository - for each repository where you need to use this. [Example](https://github.com/SonarSource/re-terraform-aws-vault/blob/3f5efb03a94a38d1346cbde62631ce1a340b5e14/orders/bubble-dotnet.yaml#L180-L182)

## Example usage

```yaml
name: Toggle lock branch

on:

>   workflow_dispatch:    # Triggered manually from GitHub UI / Actions

Reviewer: ```suggestion
  workflow_dispatch:    # Triggered manually from the GitHub UI / Actions
```
